File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int count = 0;
        ListNode* ptr = head;
        while(ptr != nullptr){
            count++;
            ptr = ptr->next;
        }
        if(count==n){
            ListNode* d = head;
            head = head->next;
            delete d;
            return head;
        }
        ptr = head;
        int i = count-n-1;
        while(i--){
            ptr = ptr->next;
        }
        ListNode* d = ptr->next;
        ptr->next = ptr->next->next;
        delete d;
        return head;

    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    ListNode reverse(ListNode head){

        ListNode prev = null,ptr = head,next = null;
        while(ptr != null){
            next = ptr.next;
            ptr.next = prev;
            prev = ptr;
            ptr = next;
        }
        return prev;
    }
    public ListNode removeNthFromEnd(ListNode head, int n) {
        
        head = reverse(head);
        if(n==1){
            head = head.next;
        }
        else{
            ListNode ptr = head;
            for(int i=1 ; i<n-1 ; i++){
                ptr = ptr.next;
            }
            ptr.next = ptr.next.next;
        }
        head = reverse(head);
        return head;

        // while(ptr != null){
        //     next = ptr.next;
        //     ptr.next = prev;
        //     prev = ptr;
        //     ptr = next;
        // }
        // return prev;
        
    }
}